x/distribution/keeper/allocation.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// Allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateFees(ctx sdk.Context) {
+	ctx.Logger().With("module", "x/distribution").Error(fmt.Sprintf("allocation height: %v", ctx.BlockHeight()))
+
+	// if there is no power in the system nothing should be allocated
+	bondedTokens := k.stakeKeeper.TotalPower(ctx)
+	if bondedTokens.IsZero() {
+		return
+	}
+
+	// get the proposer of this block
+	proposerConsAddr := k.GetProposerConsAddr(ctx)
+	proposerValidator := k.stakeKeeper.ValidatorByConsAddr(ctx, proposerConsAddr)
+	proposerDist := k.GetValidatorDistInfo(ctx, proposerValidator.GetOperator())
```

Comment:
This is a bit inefficient since `ValidatorByConsAddr` looks up the operator address, *then* looks up the validator

Reply:
I don't see a better alternative, we need to store by cons address, no other option AFAICT